spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsMessageConversionDelegate.java

```diff
@@ -91,18 +91,20 @@ public KStream serializeOnOutbound(KStream<?, ?> outboundBindTarget) {
 		MessageConverter messageConverter = this.compositeMessageConverter;
 		final PerRecordContentTypeHolder perRecordContentTypeHolder = new PerRecordContentTypeHolder();
 
-		final KStream<?, ?> kStreamWithEnrichedHeaders = outboundBindTarget.mapValues((v) -> {
-			Message<?> message = v instanceof Message<?> ? (Message<?>) v
-					: MessageBuilder.withPayload(v).build();
-			Map<String, Object> headers = new HashMap<>(message.getHeaders());
-			if (!StringUtils.isEmpty(contentType)) {
-				headers.put(MessageHeaders.CONTENT_TYPE, contentType);
-			}
-			MessageHeaders messageHeaders = new MessageHeaders(headers);
-			final Message<?> convertedMessage = messageConverter.toMessage(message.getPayload(), messageHeaders);
-			perRecordContentTypeHolder.setContentType((String) messageHeaders.get(MessageHeaders.CONTENT_TYPE));
-			return convertedMessage.getPayload();
-		});
+		final KStream<?, ?> kStreamWithEnrichedHeaders = outboundBindTarget
+			.filter((k, v) -> v != null)
```

Comment:
I don't think we should filter these; they may be sending a tombstone record.

`null` values should become `KafkaNull.INSTANCE`.

Reply:
The binder sends the `null` value down and then in the serialization method simply skip the serialization by the presence of this filer. See [here](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/blob/master/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KStreamBinder.java#L143).